packages/preferences/src/browser/preferences-widget.ts

```diff
@@ -0,0 +1,237 @@
+    @inject(PreferenceProvider) @named(PreferenceScope.User) protected readonly userPreferenceProvider: UserPreferenceProvider;
+
+    constructor() {
+        super('theia-user-preferences-container', 'User Preferences', USER_PREFERENCES_TREE_WIDGET_ID);
+    }
+
+    async getEditor(): Promise<EditorWidget | undefined> {
+        return await this.widgetManager.getOrCreateWidget("code-editor-opener", this.userPreferenceProvider.getUri().withoutFragment().toString()) as EditorWidget;
+    }
+
+}
+
+export class WorkspacePreferencesContainer extends PreferencesContainer {
+
+    @inject(WidgetManager) protected readonly widgetManager: WidgetManager;
+    @inject(PreferenceProvider) @named(PreferenceScope.Workspace) protected readonly workspacePreferenceProvider: WorkspacePreferenceProvider;
+
+    constructor() {
+        super('theia-workspace-preferences-container', 'Workspace Preferences', WORKSPACE_PREFERENCES_TREE_WIDGET_ID);
+    }
+
+    async getEditor(): Promise<EditorWidget | undefined> {
+        const wsUri = await this.workspacePreferenceProvider.getUri();
+        if (wsUri) {
+            return await this.widgetManager.getOrCreateWidget("code-editor-opener", wsUri.withoutFragment().toString()) as EditorWidget;
+        }
+    }
+}
+
+export abstract class PreferencesTreeWidget extends TreeWidget {
+
+    protected abstract handlePropertySelection: (property: string, value: any) => void;
+    private preferencesGroupNames: string[] = [];
+    private readonly properties: { [name: string]: PreferenceProperty };
+
+    @inject(PreferencesMenuFactory) protected readonly preferencesMenuFactory: PreferencesMenuFactory;
+    @inject(PreferenceService) protected readonly preferenceService: PreferenceService;
+
+    protected constructor(@inject(TreeModel) readonly model: TreeModel,
+                          @inject(TreeProps) protected readonly treeProps: TreeProps,
+                          @inject(PreferencesDecorator) protected readonly decorator: PreferencesDecorator,
+                          @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer,
+                          @inject(PreferenceSchemaProvider) protected readonly preferenceSchemaProvider: PreferenceSchemaProvider) {
+        super(treeProps, model, contextMenuRenderer);
+        this.addClass('theia-preferences');
+
+        this.id = "theia-preferences-tree";
+
+        this.properties = this.preferenceSchemaProvider.getCombinedSchema().properties;
+        for (const property in this.properties) {
+            if (property) {
+                const group: string = property.substring(0, property.indexOf('.'));
+                if (this.preferencesGroupNames.indexOf(group) < 0) {
+                    this.preferencesGroupNames.push(group);
+                }
+            }
+        }
+    }
+
+    protected onAfterAttach(msg: Message): void {
+        this.initializeModel();
+        super.onAfterAttach(msg);
+    }
+
```

Comment:
I think in this case it would make sense to allow using a simple click to trigger the same action.

Reply:
I mean I would like to open the 'value options' with the left click.

Reply:
done